### ISSUE TITLE
Fix the card matching for Coolkey and Muscle drivers

### ISSUE THREAD

**Issue by Jakuje** [OpenSC/OpenSC#1483]
### Problem Description

In reference to the comment https://github.com/OpenSC/OpenSC/issues/1377#issuecomment-394778180 from @dengert  I just talked with Bob and this is an issue of some Java cards that can not be handled in the applet itself. The SELECT APDU is handled by the Card itself so the applet can not do anything about it. This used to be an issue also for CAC or PIV cards and they no longer depend just on the return code from SELECT APDU. 

### Proposed Resolution

I will try to propose a patch for Coolkey driver soon so it can behave better, avoid mismatches and therefore moved above muscle. I do not know quite much anything about muscle nor I have the applet so I can not help in this way, but if the muscle is considered as active, I assume some muscle developers can do the same for muscle.

### Steps to reproduce

Soon after boot (WaitForSlotEvent + plugging the card in) or when running several sessions soon after each other from (p11test), the SELECT APDU returns 90 00 regardless the applet is actually available on the card. This can mismatch coolkey card as muscle or vice versa.

**Comment by dengert** [comment]
Is there any additional command to a muscle or coolkey to determine which
applet is active?

After a select aid all drivers shoild check for response with correct AID.
If no data returned muscle and coolkey drivers need to check for something
else.

**Comment by dengert** [comment]
For example is there any Java card command that could return what applets
are on the card?

Or does muscle or coolkey respond. To 7E command?

**Comment by Jakuje** [comment]
> Or does muscle or coolkey respond. To 7E command?

I did not find any of the applets explicitly responding to this instruction.

Reading through the source code of both applet, I noticed that the GetStatus has a difference in handling the P1:

 * The [coolkey applet](https://github.com/dogtagpki/coolkey/blob/master/applet/src/com/redhat/ckey/applet/CardEdge.java#L1333) has this argument reserved for call to `GetMemory()`, which is not implemented and returns invalid instruction (`0x6D00`)
* The [muscle applet](https://github.com/martinpaljak/MuscleApplet/blob/d005f36209bdd7020bac0d783b228243126fd2f8/src/com/musclecard/CardEdge/CardEdge.java#L2115) requires this argument is set to zero, otherwise it returns incorrect P1 (`0x9C10`)

This should allow as to make coolkey matching more robust and move the driver a bit up.

The muscle driver already tries to call the get status during matching, which should prevent mismatching against different cards (sc-hsm, ...), but it actually did not prevent from matching against Coolkey, which is using the same instruction.

See PR #1500 for proposed patch.
